src/hooks/useExec.ts

```diff
@@ -239,19 +252,19 @@ export async function which(arg0: string | undefined) {
             // we are being executed via the command not found handler inside a dev-env
             // so let’s use the version that was already calculated for this dev-env
             if ("version" in inenv) {
-              found = {...inenv, constraint: new semver.Range(`=${inenv.version}`), shebang: provider }
+              found = {...inenv, constraint: new semver.Range(`=${inenv.version}`), shebang: provider, precmd: undefined }
             } else {
-              found = {...inenv, shebang: provider }
+              found = {...inenv, shebang: provider, precmd: undefined }
```

Comment:
not required surely since you defined the type as `precmd?: string`

Reply:
TS was complaining. I'll take another look.